### DISCUSSION TITLE
Endpoint for sensor JSON data while WebServer in user mode?

### DISCUSSION THREAD

**Discussion by bartoszbielawski** [arendst/Tasmota#16035] [Q&A] [answered]
Hi, 

I wanted to deploy some ESP8266 with Tasmota equipped with some sensors.
Because the network to which devices are connected is widely available I'd like to switch WebServer to user mode (mode 1).

To extract data from sensors I need to use HTTP GET. Is there an endpoint that serves JSON data (something Like Status 10) while the WebServer is in user mode?

I tried looking at the code and I found a place where the response is generated, it's in the HTTP server driver, in HandleRootStatusRefresh, but I'm not sure if the new code could be plugged there.

Any ideas?

Cheers,
Bartosz

**Comment by sfromis** [comment, marked as answer]
No, command access is not available in user mode. No other standard features for grabbing JSON via http. Tasmota is designed for MQTT, that's the "M" in the name. If you used ESP32, you'd be able to setup a web server using Berry.

One way could be to make rules to capture the sensor values you want, and use WebQuery to contact a server you control, including the readings.

**Comment by bartoszbielawski** [comment]
I understand there's no command access. My point is that "Status 10" command is almost the same as /?m=1 in user mode. It's the same data, just in a different format. I'm looking for an endpoint that would let me pull the data available via /?m=1 but with JSON - without issuing any commands. Of course a new endpoint could be added that would serve this purpose - I unfortunately don't know the code yet enough.

**Reply by barbudor** [reply]
Beware that '/?m=1' is an internal, undocumented, unsupported web call only intended for Tasmota internals.
It's behavior or even its existence has no guarantee across release.

**Reply by sfromis** [reply]
I already told you "no other standard features for grabbing JSON via http", and what other strategies could work. You might also be able to do something with the scripting language on ESP8266, but I'm without much knowledge there.

**Reply by bartoszbielawski** [reply]
@barbudor Thanks, I looked at the code again and now I probably know how to modify ?m=1, but if this is an internal API then there's no point in adding things there. Is there another endpoint that is meant to be used for extensions?

@sfromis I understand, there are not features like this right now. I could add one since the data clearly is available. 
I could also just pull data using the current endpoint and parse it myself - but this is more fragile than having a dedicated endpoint.

**Reply by Jason2866** [reply]
No, the official "endpoint" is mqtt, The "M" in TASMOTA stands for mqtt. A PR which is bloating the code will not be integrated.

**Reply by bartoszbielawski** [reply]
Prebuilt binaries for ESP8266 don't come with MQTT TLS so I will have to compile the code myself anyway. Thanks!

**Comment by barbudor** [comment]
It just came back to my memory that the Prometeus driver (USE_PROMETHEUS) has a '/metrics' endpoint which returns a combination of STATE and SENSORS
It is meant to be polled by a Prometeus server so might work also for your use case
